src/store.js

```diff
@@ -0,0 +1,110 @@
+import { remoteRead } from './api';
+import Registry from './registry';
+import defineModel from './model';
+
+/**
+ * @module
+ * @ignore
+ */
+export const missingStore = () => (
+  new Error(`
+    Active-Redux has not yet been bound to a store:
+
+    import ActiveRedux from 'active-redux';
+    import myStore from './store';
+
+    ActiveRedux.bind(myStore);
+  `)
+);
+
+// Used for querying the store #where / #find
+const deepPartialEqual = (obj1, obj2) => (
+  Object.entries(obj1).reduce((equal, [key, value]) => {
+    if (equal === false) return equal;
+    if (obj2 === undefined) return false;
+
+    if (typeof value === 'object' && !Array.isArray(value)) {
+      return deepPartialEqual(value, obj2[key]);
+    }
+
+    if (value !== obj2[key]) return false;
+
+    return equal;
+  }, true)
+);
+
+// Query through an array of objects
+export const queryData = (entities, query) => (
+  entities.filter((entity) => (
+    Object.entries(query).reduce((match, [queryKey, queryValue]) => {
+      if (match === false) return match;
+
+      const entityValue = entity[queryKey];
+
+      if (Array.isArray(queryValue)) {
+        return queryValue.includes(entityValue);
+      }
+      if (typeof queryValue === 'object') {
+        return deepPartialEqual(queryValue, entityValue);
+      }
+      return queryValue === entityValue;
+    }, true)
+  ))
+);
+
+// Instantiate JSON-API data with its corresponding Model
+const initModel = (Model) => (data) => new Model(data);
+
+class Store {
```

Comment:
This feels a lot cleaner than our distinctions before. I found this split while testing / looking at dependencies coming into the tests. The store felt uncomfortable to test. It's a lot easier to do so now.